### ISSUE TITLE
Training problem

### ISSUE THREAD

**Issue by nuitvolgit** [zzh8829/yolov3-tf2#70]
I use tensorflow 2.0.0-rc1

I trained using train.py with a default training image: "girl.png"
After training with FLAGS.mode='fit', FLAGS.transfer='none', 
I tested it with the same trained image using detect.py
But I got following warnings and couldn't detect any objects as well.

.......
WARNING:tensorflow:Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-0.kernel
W0919 11:04:29.340978 140067676804928 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-0.kernel
WARNING:tensorflow:Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-1.gamma
W0919 11:04:29.341034 140067676804928 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-1.gamma
WARNING:tensorflow:Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-1.beta
W0919 11:04:29.341070 140067676804928 util.py:144] Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-1.beta
WARNING:tensorflow:Unresolved object in checkpoint: (root).optimizer's state 'm' for (root).layer_with_weights-0.layer_with_weights-2.kernel
................... and so on..

It looks like weights are not fully restored for some reason.

If I train with FLAGS.mode = 'eager_tf', I don't get such warning when running detect.py

It seems like saving weight using "model.save_weights" does not provide warning but
using "ModelCheckpoint" does provide warning.

I am not sure.

Any ideas?

**Comment by nivsmall** [comment]
I had the same issue before,my issue was that the imported tensorflow was version 1 and not 2.0 (even though the V.2 was installed in the environment)
did you verify correct version during run?

**Comment by nuitvolgit** [comment]
I still get the same warning but the detection works.
It turned out that the reason of detection failure was not related to the warnings.
I tried to train a single class detection but the class probability was not trained since the softmax for a single class always gives 1.
So I changed the score value in the yolo_nms function such that if the class number is one, then score is the same as the confidence.

**Comment by mmortazavi** [comment]
I am having a quick start using this repo. Just read your comments about single class detection! How did you change in score value in the yolo_nms function? Would you mind sharing your modified yolo_nms function here so we could use as well? Thanks

**Comment by nuitvolgit** [comment]
I changed the following line in yolo_nms function:
==> scores = confidence * class_probs

to like this:

==> if classes > 1:
          scores = confidence * class_probs
      else:
          scores = confidence

This is because the class_prob is not trained for a single class case.
So its value is almost like zero from the start to the end of the training.

**Comment by nuitvolgit** [comment]
So I couldn't fix the warning but everything works fine so I decided to
ignore the warning.
Thanks.

2019년 9월 20일 (금) 오전 2:55, nivsmall <notifications@github.com>님이 작성:

**Comment by SpontaneousDuck** [comment]
This fix for one class should be pushed up to the main! Maybe i'll get around to a PR

**Comment by lazerliu** [comment]
My detection is also not work,I train with pascal 16 epochs,and then detect nothing! 
I wonder what the correct steps to validate the model I trained.thks

**Comment by zzh8829** [comment]
@lazerliu see this tutorial https://github.com/zzh8829/yolov3-tf2/blob/master/docs/training_voc.md
